### DISCUSSION TITLE
Thumbnail Not working

### DISCUSSION THREAD

**Discussion by mosheas** [Jyers/Marlin#1796] [Q&A]
Use firmware 2.0.2h 10x10, (on printer set box set Thumbnail on), 
![IMG_20220118_102307](https://user-images.githubusercontent.com/32372234/149902668-d84a3709-f0cd-4019-9b86-bd400ae2c517.jpg)
Cura version 4.13
Although I set up in Cura to add this thumbnail is not displayed on the printer screen. 
In the G-code file, I see the addition of the thumbnail code,and yet it's not on the screen. 
 I'm missing something?

**Comment by K2Kevin** [comment]
Try the plugin found within https://github.com/Jyers/Marlin/files/7808535/Marlin-bugfix-JyersUI-v202h.zip

in this folder : Slicer post-processing scripts for Gcode Preview

**Reply by mosheas** [reply]
Thank you, looks great.

**Reply by Sonicboom247** [reply]
can this be tested with python 3.10.2? looks like 3.9.10 and 3.10.2 were released on the same day

**Reply by K2Kevin** [reply]
You may want to hold off using the Plugin option with this firmware release (Unless you have only a few prints stored on the SDCard).  The list becomes VERY slow to scroll, literally like 1 second each item as you scroll down you list.  I had a card with about 50 prints, took a minute or more to get to the bottom of the list.  Wouldn't it be nice if the list kept items in order (I often find items not in Alphabetic order) or god forbid a sort feature (New on top).

**Reply by Sonicboom247** [reply]
ahh, i have mine set to NOT sort alphabetically but newest at the top with #define SDCARD_RATHERRECENTFIRST but even that is hit or miss.  most i have on an SDcard is maybe 15-20, then all of a sudden it gets corrupted even using eject from PC.

**Comment by Sonicboom247** [comment]
anyone else seeing this error with python 3.10.2
Post-processing script "C:\Program Files\Python310\python.exe" "C:\Program Files\Prusa3D\Scripts_Post-processings\PrusaSlicer_JPEG_Preview.py" on file C:\Users\THREED~1\AppData\Local\Temp\.9924.gcode.pp failed.
Error code: 1

**Reply by Sonicboom247** [reply]
just tested with the new exe, same errors or it can't find it even if exe is in same directory as slicer
![Prusa1](https://user-images.githubusercontent.com/84334417/153409311-c32cf1c2-e5c7-4efd-8c5d-8a8d024884b8.jpg)
![Prusa2](https://user-images.githubusercontent.com/84334417/153409321-a95a1778-8538-46ed-9a2b-06efef3965d7.jpg)
![prusa3](https://user-images.githubusercontent.com/84334417/153409329-6b6dc621-6e0f-439e-a1cb-b6e25d1f5a22.jpg)
![prusa4](https://user-images.githubusercontent.com/84334417/153409353-a2a56d4a-073a-4f87-90aa-e2cd57a1cad4.jpg)

**Reply by Sonicboom247** [reply]
@tititopher68-dev 
using latest Version 2.4.0+win64, also tested Cura 4.13, no error in Cura but also no thumbnail.
tested 3 different ways and all give error, in my first screenshot i did test using _P24
"C:\Program Files\Python310\python.exe" "C:\Program Files\Prusa3D\PrusaSlicer_JPEG_Preview_P24.py"
C:\Program_Files\Python310\python.exe C:\Program_ Files\Prusa3D\PrusaSlicer_JPEG_Preview_P24.py
PrusaSlicer_JPEG_Preview_P24.py
![prusa5](https://user-images.githubusercontent.com/84334417/153435182-02476279-655e-4dd3-9bc9-2ef97486a1a8.jpg)

**Reply by Sonicboom247** [reply]
> But what I don't like is your tilde (????) in your path and the structure itself which should be `'...\ AppData\Local\Temp\.PrusaSlicer.upload.....9924.gcode.pp'`
> 
> I think you have a Config issue with PrusaSlicer and your PC.

you might be correct as i do test their beta versions. reinstalling and testing fresh install

**Reply by Sonicboom247** [reply]
i will try this as well, your readme instructions show Python 3.10 so i downloaded 3.10 or 3.10.1, i had tested same with 3.9, the tilde is just Windows way of showing a shortened path, not sure why as it only has 3 more letters, ThreeDCNC..

**Reply by Sonicboom247** [reply]
> @Sonicboom247 Concerning your shorted username, this can happen when WIndows installs with a Microsoft account an it cannot connect to learn the Display Name so it truncates the email address first five letters. I escalated this to developers several years ago and heard back they did not plan to change it. Following this method to solve this issue : [Username shorted issue in Windows10](https://www.windowscentral.com/how-change-account-name-windows-10-sign-screen)

interesting, yes i confirmed, mine is correct in registry and in any other programs i installed have no issues, Blender, Vcarve Pro, Lychee Slicer, Cura, Lightburn, etc..
![profile](https://user-images.githubusercontent.com/84334417/153448837-f23b536f-77e3-4146-9be9-ba55307fae74.jpg)

this is why there is a tilde, Prusa is simply placing the temp gcode in a temp location
Script execution
"Each script will be passed the absolute path of a temporary G-code file that PrusaSlicer generates. This file is stored in a temporary folder on your drive (typically your fast system drive). The script is then executed to modify the G-code in place and the resulting G-code file is written to your selected target folder"

I did notice i did not have the correct path to include \Scripts_Post-processings, but after fixing it i still get an error.
Post-processing script "C:\Program Files\Python310\python.exe" "C:\Program Files\Prusa3D\Scripts_Post-processings\PrusaSlicer_JPEG_Preview_P24.py" on file C:\Users\THREED~1\AppData\Local\Temp\.14236.gcode.pp failed.
Error code: 1

**Reply by Sonicboom247** [reply]
ok it is working now!, i used your most updated instructions from the link instead of the readme included.  I copied the PrusaSlicer_JPEG_Preview_P24 exe to inside PrusaSlicer directory and only placed PrusaSlicer_JPEG_Preview_P24.exe in output option for post processing script.  i think i was getting mixed up with the readme and web. 

thank you for your patience as always!

**Comment by Sonicboom247** [comment]
so just to be 100% up to date, i decided to update the firmware from 2.0.2H to 2.0.2i, i have a 4.2.2 A board so i selected the V202I-RET6 firmware from your link and selected UBL-BLTouch-10x10 config files, i copied the two files into vs code and compiled but the screen now stays black.  i even updated the screen with a fresh GOTCHA and it blue then turned orange but screen still black after restart,

**Reply by Sonicboom247** [reply]
i went back to 2.0.2h and no problem, go back to 2.0.2.i and blank screen...:(

**Reply by Sonicboom247** [reply]
ok, i got it, it did not like me changing the baud rate to 250000 which is odd, i had 2.0.2h running @ 250000

**Reply by Sonicboom247** [reply]
ok, i am going to have to play with this for a while or just compile fresh with my 2.0.2h settings, tried to home and printer kill error unable to home, so going to just go through line by line.

**Reply by Sonicboom247** [reply]
you are amazing! thank You!, reset to default fixed homing and i have all jpeg thumbnails and fresh new firmware!, thanks for your help 1million times, if you need any testing feel free to ask me, now go get some sleep!lol
